lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -5134,6 +5134,14 @@ Op LLVMToSPIRVBase::transBoolOpCode(SPIRVValue *Opn, Op OC) {
 SPIRVInstruction *
 LLVMToSPIRVBase::transBuiltinToInstWithoutDecoration(Op OC, CallInst *CI,
                                                      SPIRVBasicBlock *BB) {
+  if (BM->isAllowedToUseVersion(VersionNumber::SPIRV_1_4)) {
```

Comment:
Why is this conditional?  We should _never_ generate OpAtomicCompareExchangeWeak for _any_ SPIR-V version.

(Can we map the OpenCL C atomic_compare_exchange_weak builtin to OpAtomicCompareExchange before we even get here, and remove this code?)

Reply:
@MrSidims @StuartDBrady. I've updated to remove the conditional so it's always the case the OpAtomicCompareExchange is used for any SPIR-V version. Please let me know if this is enough to resolve the issue.

Reply:
Regarding a mapping, an earlier patch used that approach & had a range of issues pop-up (e.g. failing transcoding testing - atomics.spt) where atomic_compare_exchange_strong_explicit(..) would be generated with unexpected input parameter types and so that approach might require deeper investigation of that issue.

Reply:
@StuartDBrady should be addressed by this point